noir_stdlib/src/hash/poseidon.nr

```diff
@@ -3,66 +3,146 @@ use crate::field::modulus_num_bits;
 use crate::hash::Hasher;
 use crate::default::Default;
 
-struct PoseidonConfig<M,N> {
+struct PoseidonConfig<M, N, X> { 
     t: Field, // Width, i.e. state size
     rf: u8, // Number of full rounds; should be even
     rp: u8, // Number of partial rounds
     alpha: Field, // S-box power; depends on the underlying field
-    ark: [Field; M], // Additive round keys
-    mds: [Field; N] // MDS Matrix in row-major order
+    C: [Field; N],
+    Mat: [[Field;M]; M], // Additive round keys
+    PMat: [[Field;M]; M],
+    S: [Field; X],
```

Comment:
```suggestion
    Mat: [[Field; M]; M], // Additive round keys
    PMat: [[Field; M]; M],
    S: [Field; X],
```
Nit

Also it would be nice to have comments on `C`, `PMat`, and `S` as well

Reply:
Have asked Zac to comment with some explanation/better naming for these fields.